redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/protocolHostGeneration.kt

```diff
@@ -115,6 +129,37 @@ internal fun generateProtocolFactory(
             .initializer("mismatchHandler")
             .build(),
         )
+        .addProperty(
+          PropertySpec.builder(
+            "childrenTags",
+            MAP.parameterizedBy(WidgetTag, LIST.parameterizedBy(ChildrenTag)),
+            OVERRIDE,
+          )
+            .initializer(
+              buildCodeBlock {
+                val mapOf = MemberName("kotlin.collections", "mapOf")
+                val listOf = MemberName("kotlin.collections", "listOf")
+                add("%M(⇥\n", mapOf)
+                for (dependency in schemaSet.all.sortedBy { it.widgets.firstOrNull()?.tag ?: 0 }) {
+                  for (widget in dependency.widgets.sortedBy { it.tag }) {
+                    add("%T(%L) to %M(", WidgetTag, widget.tag, listOf)
+                    var first = true
+                    for (children in widget.traits
+                      .filterIsInstance<ProtocolChildren>()
+                      .sortedBy { it.tag }
+                    ) {
+                      if (!first) add(", ")
+                      first = false
+                      add("%T(%L)", ChildrenTag, children.tag)
+                    }
```

Comment:
If you want, this can be
```suggestion
                    add(
                      widget.traits
                        .filterIsInstance<ProtocolChildren>()
                        .sortedBy { it.tag }
                        .map { CodeBlock.of("%T(%L)", ChildrenTag, children.tag) }
                        .joinToCode()
                    )
```

Reply:
Filed an issue to collapse the `map`+`joinToCode` into one line, too: https://github.com/square/kotlinpoet/issues/1873

Reply:
Done.